sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/KeyEncryptionKeyClientBuilder.java

```diff
@@ -106,14 +128,58 @@ public KeyEncryptionKey buildKeyEncryptionKey(String keyId) {
 
         if (builder.getCredential() == null) {
             throw logger.logExceptionAsError(new IllegalStateException(
-                "Key Vault credentials are required to build the key encryption key async client"));
+                "Key Vault credentials cannot be null and are required to build the key encryption key async client"));
         }
 
         HttpPipeline pipeline = builder.setupPipeline();
 
         return Mono.defer(() -> Mono.just(new KeyEncryptionKeyAsyncClient(keyId, pipeline, serviceVersion)));
     }
 
+    /**
+     * Creates a {@link KeyEncryptionKeyAsyncClient} based on options set in the builder.
+     * Every time {@code buildAsyncKeyEncryptionKey(String)} is called, a new instance of
+     * {@link KeyEncryptionKeyAsyncClient} is created.
+     *
+     * <p>If {@link KeyEncryptionKeyClientBuilder#pipeline(HttpPipeline) pipeline} is set, then the {@code pipeline}
+     * and {@code keyId} are used to create the {@link KeyEncryptionKeyAsyncClient async client}.
+     * All other builder settings are ignored. If {@code pipeline} is not set, then
+     * ({@link KeyEncryptionKeyClientBuilder#credential(TokenCredential) jsonWebKey vault credential} and
+     * {@code keyId} are required to build the {@link KeyEncryptionKeyAsyncClient async client}.</p>
+     *
+     * @param key the key to be used for crypto operations.
+     *
+     * @return A {@link KeyEncryptionKeyAsyncClient} with the options set from the builder.
+     * @throws IllegalStateException If {@link KeyEncryptionKeyClientBuilder#credential(TokenCredential)} or
+     * {@code key} have not been set.
+     */
+    public Mono<? extends AsyncKeyEncryptionKey> buildAsyncKeyEncryptionKey(KeyVaultKey key) {
```

Comment:
Should this method be named `buildAsyncKeyEncryptionKeyClient`?

Reply:
This is used by other SDKs which utilize AsyncKEK and KEK.
So, from user perspective, they will use KV to build what other SDKs consume, 
this will be straight to the point and less confusing.